include/wasm/WasmModule.h

```diff
@@ -169,20 +172,19 @@ class WasmModule
     int threadPoolSize = 0;
     std::vector<uint32_t> threadStacks;
 
-    std::shared_mutex moduleMemoryMutex;
-    std::mutex modulePthreadsMutex;
-    std::mutex moduleStateMutex;
```

Comment:
It was quite hard to reason about these mutexes, so I tried to simplify with the `moduleMutex` (and one other object-specific mutex for the `sharedMemWasmPtrs`.